<Type Name="UnhandledExceptionAction" FullName="System.Activities.UnhandledExceptionAction">
  <Metadata><Meta Name="ms.openlocfilehash" Value="90beb3de2eb7f379bd39d06382a58e66536329a0" /><Meta Name="ms.sourcegitcommit" Value="434f60616a9793fa8436744549fc856e94f7a648" /><Meta Name="ms.translationtype" Value="HT" /><Meta Name="ms.contentlocale" Value="es-ES" /><Meta Name="ms.lasthandoff" Value="08/24/2018" /><Meta Name="ms.locfileid" Value="36716528" /></Metadata><TypeSignature Language="C#" Value="public enum UnhandledExceptionAction" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed UnhandledExceptionAction extends System.Enum" />
  <TypeSignature Language="DocId" Value="T:System.Activities.UnhandledExceptionAction" />
  <TypeSignature Language="VB.NET" Value="Public Enum UnhandledExceptionAction" />
  <TypeSignature Language="C++ CLI" Value="public enum class UnhandledExceptionAction" />
  <TypeSignature Language="F#" Value="type UnhandledExceptionAction = " />
  <AssemblyInfo>
    <AssemblyName>System.Activities</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Enum</BaseTypeName>
  </Base>
  <Docs>
    <summary><span data-ttu-id="69149-101">Especifica la acción que se produce cuando escapa una excepción de la raíz de un flujo de trabajo.</span><span class="sxs-lookup"><span data-stu-id="69149-101">Specifies the action that occurs when an exception escapes the root of a workflow.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69149-102">La función <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> se invoca si una excepción escapa de la raíz del flujo de trabajo.</span><span class="sxs-lookup"><span data-stu-id="69149-102">The <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> function is invoked if an exception escapes the root of the workflow.</span></span> <span data-ttu-id="69149-103">La clase <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> proporciona acceso a la excepción, así como un puntero a la clase <xref:System.Activities.Activity> que generó la excepción.</span><span class="sxs-lookup"><span data-stu-id="69149-103">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> provides access to the exception as well as a pointer to the <xref:System.Activities.Activity> that generated the exception.</span></span> <span data-ttu-id="69149-104">Terminate es la acción predeterminada si no hay ningún <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> se especificó un controlador.</span><span class="sxs-lookup"><span data-stu-id="69149-104">Terminate is the default action if no <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is specified.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="69149-105">En el ejemplo siguiente se invoca un flujo de trabajo que produce una excepción.</span><span class="sxs-lookup"><span data-stu-id="69149-105">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="69149-106">El flujo de trabajo no controla la excepción y se invoca el controlador de la propiedad <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>.</span><span class="sxs-lookup"><span data-stu-id="69149-106">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="69149-107">El objeto <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> se inspecciona para proporcionar información acerca de la excepción y se termina el flujo de trabajo.</span><span class="sxs-lookup"><span data-stu-id="69149-107">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is terminated.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#6](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#6)]  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <Member MemberName="Abort">
      <MemberSignature Language="C#" Value="Abort" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Abort = int32(0)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Abort" />
      <MemberSignature Language="VB.NET" Value="Abort" />
      <MemberSignature Language="C++ CLI" Value="Abort" />
      <MemberSignature Language="F#" Value="Abort = 0" Usage="System.Activities.UnhandledExceptionAction.Abort" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>0</MemberValue>
      <Docs>
        <summary><span data-ttu-id="69149-108">Especifica que la clase <see cref="T:System.Activities.WorkflowApplication" /> debe anular el flujo de trabajo.</span><span class="sxs-lookup"><span data-stu-id="69149-108">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should abort the workflow.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69149-109">Esto hace que se llame a la propiedad <xref:System.Activities.WorkflowApplication.Aborted%2A> cuando se complete el proceso de anulación.</span><span class="sxs-lookup"><span data-stu-id="69149-109">This results in <xref:System.Activities.WorkflowApplication.Aborted%2A> being called when the abort process is complete.</span></span> <span data-ttu-id="69149-110">Como razón de la anulación se usa la excepción no controlada.</span><span class="sxs-lookup"><span data-stu-id="69149-110">The unhandled exception is used as the abort reason.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="69149-111">En el ejemplo siguiente se invoca un flujo de trabajo que produce una excepción.</span><span class="sxs-lookup"><span data-stu-id="69149-111">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="69149-112">El flujo de trabajo no controla la excepción y se invoca el controlador de la propiedad <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>.</span><span class="sxs-lookup"><span data-stu-id="69149-112">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="69149-113">La clase <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> se inspecciona para proporcionar información sobre la excepción y se anula el flujo de trabajo.</span><span class="sxs-lookup"><span data-stu-id="69149-113">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is aborted.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#7](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#7)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Cancel">
      <MemberSignature Language="C#" Value="Cancel" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Cancel = int32(1)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Cancel" />
      <MemberSignature Language="VB.NET" Value="Cancel" />
      <MemberSignature Language="C++ CLI" Value="Cancel" />
      <MemberSignature Language="F#" Value="Cancel = 1" Usage="System.Activities.UnhandledExceptionAction.Cancel" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>1</MemberValue>
      <Docs>
        <summary><span data-ttu-id="69149-114">Especifica que la clase <see cref="T:System.Activities.WorkflowApplication" /> debe programar la cancelación de la actividad raíz y reanudar la ejecución.</span><span class="sxs-lookup"><span data-stu-id="69149-114">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should schedule the cancellation of the root activity and resume execution.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69149-115">Esto hace que se llame a <xref:System.Activities.WorkflowApplication.Completed%2A> una vez completado el proceso de cancelación.</span><span class="sxs-lookup"><span data-stu-id="69149-115">This results in <xref:System.Activities.WorkflowApplication.Completed%2A> being called when the cancellation process is complete.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="69149-116">En el ejemplo siguiente se invoca un flujo de trabajo que produce una excepción.</span><span class="sxs-lookup"><span data-stu-id="69149-116">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="69149-117">El flujo de trabajo no controla la excepción y se invoca el controlador de la propiedad <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>.</span><span class="sxs-lookup"><span data-stu-id="69149-117">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="69149-118">La clase <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> se inspecciona para proporcionar información sobre la excepción y se cancela el flujo de trabajo.</span><span class="sxs-lookup"><span data-stu-id="69149-118">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is cancelled.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#8](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#8)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Terminate">
      <MemberSignature Language="C#" Value="Terminate" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Terminate = int32(2)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Terminate" />
      <MemberSignature Language="VB.NET" Value="Terminate" />
      <MemberSignature Language="C++ CLI" Value="Terminate" />
      <MemberSignature Language="F#" Value="Terminate = 2" Usage="System.Activities.UnhandledExceptionAction.Terminate" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>2</MemberValue>
      <Docs>
        <summary><span data-ttu-id="69149-119">Especifica que la clase <see cref="T:System.Activities.WorkflowApplication" /> debe programar la terminación de la actividad raíz y reanudar la ejecución.</span><span class="sxs-lookup"><span data-stu-id="69149-119">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should schedule termination of the root activity and resume execution.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="69149-120">Esto hace que se llame a <xref:System.Activities.WorkflowApplication.Completed%2A> una vez finalizado el proceso de terminación.</span><span class="sxs-lookup"><span data-stu-id="69149-120">This results in <xref:System.Activities.WorkflowApplication.Completed%2A> being called when the termination process is complete.</span></span> <span data-ttu-id="69149-121">Como razón de la finalización se usa la excepción no controlada.</span><span class="sxs-lookup"><span data-stu-id="69149-121">The unhandled exception is used as the termination reason.</span></span> <span data-ttu-id="69149-122">Terminate es la acción predeterminada si no hay ningún <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> se especificó un controlador.</span><span class="sxs-lookup"><span data-stu-id="69149-122">Terminate is the default action if no <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is specified.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="69149-123">En el ejemplo siguiente se invoca un flujo de trabajo que produce una excepción.</span><span class="sxs-lookup"><span data-stu-id="69149-123">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="69149-124">El flujo de trabajo no controla la excepción y se invoca el controlador de la propiedad <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>.</span><span class="sxs-lookup"><span data-stu-id="69149-124">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="69149-125">El objeto <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> se inspecciona para proporcionar información acerca de la excepción y se termina el flujo de trabajo.</span><span class="sxs-lookup"><span data-stu-id="69149-125">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is terminated.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#6](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#6)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>